forms/submission_form.py

```diff
@@ -0,0 +1,168 @@
+                    'vessel_type':     None,
+                    'bottle_count':    None,
+                    'plate_count':     None,
+                    'vessel_count':    None,
+                    'column_count':    None,
+                    'row_count':       None,
+                    'timepoint_count': None,
+                    'technique_types': [],
+                    'strains':         [],
+                    'new_strains':     [],
+                    'metabolites':     [],
+                }
+
+            )
+
+        # Check for an existing project/study and set the submission "type" accordingly:
+        self.project_id = self._find_project_id()
+        self.study_id   = self._find_study_id()
+        self.type       = self._determine_project_type()
+
+    def update_project(self, data):
+        self.type = data['submission_type']
+        self.submission.studyDesign['project'] = {
+            'name':        data['project_name'],
+            'description': data['project_description'],
+        }
+        flag_modified(self.submission, 'studyDesign')
+
+        if self.type == 'new_project':
+            self.submission.projectUniqueID = str(uuid4())
+            self.submission.studyUniqueID   = str(uuid4())
+        elif self.type == 'new_study':
+            self.submission.projectUniqueID = data['project_uuid']
+            self.submission.studyUniqueID   = str(uuid4())
+        elif self.type == 'update_study':
+            self.submission.projectUniqueID = data['project_uuid']
+            self.submission.studyUniqueID   = data['study_uuid']
+
+        self.project_id = self._find_project_id()
+        self.study_id   = self._find_study_id()
+
+    def update_strains(self, data):
+        self.submission.studyDesign['strains']     = data['strains']
+        self.submission.studyDesign['new_strains'] = data['new_strains']
+        flag_modified(self.submission, 'studyDesign')
+
+    def update_study_design(self, data):
+        study_design = self.submission.studyDesign
+
+        study_design['vessel_type'] = data.get('vessel_type', None)
+
+        study_design['bottle_count'] = data['bottle_count']
+        study_design['plate_count']  = data['plate_count']
+        study_design['column_count'] = data['column_count']
+        study_design['row_count']    = data['row_count']
+
+        if study_design['vessel_type'] == 'bottles':
+            study_design['vessel_count'] = data['bottle_count']
+        elif study_design['vessel_type'] == 'plates':
+            study_design['vessel_count'] = data['plate_count']
+
+        study_design['timepoint_count'] = data['timepoint_count']
+        study_design['technique_types'] = data['technique_types']
+        study_design['metabolites']     = data['metabolites']
+
+        self.submission.studyDesign = study_design
+        flag_modified(self.submission, 'studyDesign')
+
+    def fetch_taxa(self):
+        strains = self.submission.studyDesign['strains']
```

Comment:
This is the answer to a previous commment of mine on documenting the table, but it's becoming super clear here; so maybe ignore.